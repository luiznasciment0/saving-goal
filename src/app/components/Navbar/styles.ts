import styled from 'styled-components'
import { generateMedia } from 'styled-media-query'

const customMedia = generateMedia({
  medium: '560px'
})

export const Nav = styled.nav`
  background-color: #ffffff;
  width: 100%;
  height: 5.6rem;
  top: 0;
  padding-top: 1.5rem;

  ${customMedia.greaterThan('medium')`
    padding-bottom: 1.5rem;
  `}
`

export const Picture = styled.picture`
  margin: 1.6rem;
  ${customMedia.greaterThan('medium')`
    margin: 2rem auto 2rem 5.2rem;
    height: 3.2rem;
    width: 10rem;
  `}
`
