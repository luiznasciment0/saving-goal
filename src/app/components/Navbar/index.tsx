import { Nav, Picture } from './styles'
import Logo from 'app/assets/images/logo.png'
import Logo2x from 'app/assets/images/logo2x.png'

const Navbar = () => {
  return (
    <Nav>
      <Picture>
        <source media="(min-width: 768px)" srcSet={Logo2x} />
        <source media="(max-width: 768px)" srcSet={Logo} />
        <img src={Logo} alt="Origin Logo" />
      </Picture>
    </Nav>
  )
}

export default Navbar
