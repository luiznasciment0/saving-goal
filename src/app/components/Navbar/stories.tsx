import { Story, Meta } from '@storybook/react/types-6-0'
import Navbar from '.'

export default {
  title: 'Navbar',
  component: Navbar,
  args: {
    title: 'title default',
    description: 'description default'
  }
} as Meta

export const Basic: Story = (args) => <Navbar {...args} />
Basic.args = {
  title: 'title basic',
  description: 'description basic'
}

export const Default: Story = (args) => <Navbar {...args} />
