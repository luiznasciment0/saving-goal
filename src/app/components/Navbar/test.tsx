import { render, screen } from '@testing-library/react'
import 'jest-styled-components'

import Navbar from '.'

describe('<Navbar />', () => {
  it('should render a bigger header on desktop', () => {
    const { container } = render(<Navbar />)

    expect(screen.getByRole('navigation')).toHaveStyleRule(
      'padding-bottom',
      '1.5rem',
      {
        media: '(min-width: 560px)'
      }
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
