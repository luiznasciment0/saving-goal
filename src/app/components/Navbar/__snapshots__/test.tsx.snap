// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render a bigger header on desktop 1`] = `
.c0 {
  background-color: #ffffff;
  width: 100%;
  height: 5.6rem;
  top: 0;
  padding-top: 1.5rem;
}

.c1 {
  margin: 1.6rem;
}

@media (min-width:560px) {
  .c0 {
    padding-bottom: 1.5rem;
  }
}

@media (min-width:560px) {
  .c1 {
    margin: 2rem auto 2rem 5.2rem;
    height: 3.2rem;
    width: 10rem;
  }
}

<nav
  class="c0"
>
  <picture
    class="c1"
  >
    <source
      media="(min-width: 768px)"
      srcset="[object Object]"
    />
    <source
      media="(max-width: 768px)"
      srcset="[object Object]"
    />
    <img
      alt="Origin Logo"
      src="[object Object]"
    />
  </picture>
</nav>
`;
